src/java_bytecode/java_bytecode_parse_tree.h

```diff
@@ -67,7 +67,9 @@ class java_bytecode_parse_treet
     bool is_public, is_protected, is_private, is_static, is_final;
     annotationst annotations;
 
-    virtual void output(std::ostream &out) const = 0;
+    virtual void output(
+      const irep_idt &class_name,
```

Comment:
This seems like something the `methodt` should store rather than something every user has to pass to `output()`?

Reply:
No, this would be an antipattern; its not unusual to require context for output, say think of indentation.

Reply:
I think it is reasonable to pass this in, I also think it is reasonable for the method to know what class it belongs to FWIW.